open-api/rest-catalog-open-api.yaml

```diff
@@ -2774,6 +2920,30 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableResult:
+      type: object
+      required:
+        - plan-tasks
+      properties:
+        plan-tasks:
+          type: array
+          items:
+            $ref: '#/components/schemas/PlanTask'
+        next-page-token:
+          $ref: '#/components/schemas/PageToken'
```

Comment:
I thought that in the discussion of the originally proposal we decided not to paginate this API because it already supports breaking up work via preplan.

Why does this API need multiple ways to divide work?

Reply:
@rdblue 
So before this pagination change was made to the protocol, the assumption we were making was that for very large tables we can create multiple shards (plan tasks) and then for each plan task retrieve the list of file scan tasks. As number of shards grew this would make the number of file scan tasks per shard a reasonable number so that pagination would not be needed.

However I think there are cases where you may not be able to create this many plan tasks therefore the number of file scan tasks per shard will be very large. Thus you will need to introduce pagination at the very least for `plan` table. 

I think `preplan` I agree maybe it might be overkill to have pagination for the list of `PlanTask`, but would like to know what others think @jackye1995 @danielcweeks @amogh-jahagirdar

Reply:
I think that the `preplan` endpoint can do anything that `plan` with pagination tokens can do. It knows ahead of time how many files may be in each manifest, so it should be able to pre-limit the size of each plan task.